### DISCUSSION TITLE
Deployment of new Winget-AutoUpdate MSI with Intune

### DISCUSSION THREAD

**Discussion by dlundahl91** [Romanitho/Winget-AutoUpdate#767] [General]
Hey team WAU

We have deployed your GPO as an Intune configuration and set the desired settings there, but we want to confirm that we don't need to set the parameters during installation of the MSI file? Or must the settings also be set during installation of the MSI file in form of parameters?

We are worried that the winget-autoupdate will run auto-updates before the configuration gets applied - and that would cause troubles in our setup. Hence this topic/question 🫡

FYI we deploy our devices with Autopilot using Intune.

Thanks a ton in advance, as always!

**Comment by KnifMelti** [comment]
Hello, there!
At default GPO checks from a client runs every 15 minutes in Windows AD (in Intune I don't know).

WAU runs the settings check/import once a day via `WAU-Policies.ps1` as SYSTEM (in the Task `Winget-AutoUpdate-Policies`)
So, you could trigger the task after installation (or run the script) to make sure the settings are correct.

**Reply by Romanitho** [reply]
Small detail here: the `WAU-Policies.ps1` script runs once a day to adjust settings that cannot be changed in any other way, such as the `Winget-AutoUpdate` scheduled task. All other settings are applied during each `Winget-AutoUpdate` run.
In my opinion, there’s no need to configure settings during installation, as the policies are applied before installation during Autopilot. However, I always include the setting to enable the whitelist to ensure that if no settings are applied, the whitelist remains empty, and nothing happens. 😅

**Comment by KnifMelti** [comment]
Theres no extra work implementing the correct settings when deploying (using the GPO for changes in the future), so that's my recommendation

**Reply by Romanitho** [reply]
I agree. Just put the most important settings during install, like `USEWHITELIST=1` (if you use it) and/or `DISABLEWAUAUTOUPDATE=1` for instance. Then, manage WAU with GPO/Policies

**Comment by dlundahl91** [comment]
Thanks so much to the both of you!

This is what we set more or less in the Intune configuration - so would this install command work well?
msiexec /i "WAU.msi" /qn USERCONTEXT=1 NOTIFICATIONLEVEL=SuccessOnly UPDATESATLOGON=0 UPDATESINTERVAL=NEVER DISABLEWAUAUTOUPDATE=1 RUN_WAU=NO

In sandbox it seems to work fine and only one scheduled task is "active" by trigger and that is the policies check, which (I guess!) will make sure that it checks the configuration file for what settings are set there, and acts accordingly.. Anything I have missed?

I owe you guys a great deal of appreciation, for sure!
